cobc/parser.y

```diff
@@ -154,6 +154,19 @@ unsigned int			cobc_in_json_generate_body = 0;
 
 /* Local variables */
 
+enum inspect_rep_keyword {
+	INSPECT_REP_DEFAULT,
+	INSPECT_REP_ALL,
+	INSPECT_REP_LEADING,
+	INSPECT_REP_FIRST,
+	INSPECT_REP_TRAILING,
+};
+
+union examine_keyword {
```

Comment:
It is not "necessary" *per se*.

However, `EXAMINE` allows the additional `UNTIL FIRST` and forbids `TRAILING`; in these instances I always feel it's cleaner to have a dedicated enumeration so one can "easily" assess whether a match/switch statement covers all cases.
I originally had the codes for `cb_build_examine_*` within `parser.y`, yet I though it would be cleaner to move it there.

Regarding the approach for building trees, `INSPECT TALLYING` and `INSPECT REPLACING` do not seem to follow the same approach already (one uses the static `inspect_keyword` variable, a switch in the parser, and some hidden state in `typeck.c`, whereas the other seems builds the list of replacement operations on the go).

Reply:
agreed on the separate enumeration; the hidden state sounds strange...

But actually both are exexuting the `cb_build_tallying_...`  and `cb_build_replacing_,,,` on the go.
The question is: can you change to build the options for `EXAMINE` on the go, too?

Reply:
By "hidden_state"  I mostly mean the `inspect_fun` within `typeck.c`.

I tried to use a similar approach for `EXAMINE`, but the case of `EXAMINE … TALLYING … REPLACING BY` requires to re-use the inspected region. So it appeared simpler and cleaner to gather all this within a single piece of code. To avoid this may require either: (i) using at least one additional static variable to record the inspected region to construct the underlying replacing clause; or (ii) finding a way to "extract" the region from the previously constructed tallying clause. (i) may be workable if you're ok to add an additional static variable ; I'm not sure there's a clean way to properly implement (ii).

Reply:
I'm OK with another static variable as needed, both in parser.y and/or in typeck.c :-)
Especially if it helps to make the code for `EXAMINE` more similar to the `INSPECT` code.

for II - if you mean within the parser inside of `examine_tallying_replacing` - you can back reference to previous tokens with $-1 and friends or move the necessary part out to the place where the token is used; but I guess you mean something else

Reply:
Actually if we're ok just putting the tree construction code into the parser, the current proposal avoids any static variable (I don't remember why I though I needed one…).